### PR TITLE
StrMap#filter function added

### PR DESCRIPTION
**Pull request by mlegenhausen**

### CODE REVIEW

File: src/StrMap.ts [hunk 1]
```diff
@@ -52,6 +52,16 @@ export class StrMap<A> {
     }
     return out
   }
+  filter(p: Predicate<A>): StrMap<A> {
```
Comment (gcanti) [inline src/StrMap.ts]: tag `@since` is missing

File: src/StrMap.ts [hunk 2]
```diff
@@ -52,6 +52,16 @@ export class StrMap<A> {
...
+    const o = this.value
+    const r: { [key: string]: A } = {}
+    for (const k in o) {
+      if (p(o[k])) {
```
Comment (gcanti) [inline src/StrMap.ts]: we can avoid duplication here with a `const value = o[k]`

File: src/StrMap.ts [hunk 3]
```diff
@@ -272,6 +282,14 @@ export const pop = <A>(k: string, d: StrMap<A>): Option<[A, StrMap<A>]> => {
   return a.isNone() ? none : some(tuple(a.value, remove(k, d)))
 }
 
+/**
+ * @function
+ * @since 1.4.0
+ */
+export const filter = <A>(fa: StrMap<A>, p: Predicate<A>): StrMap<A> => {
```
Comment (gcanti) [inline src/StrMap.ts]: since there is a `filter` method, we can avoid exporting a static function here, WDYT?

Comment (mlegenhausen) [reply src/StrMap.ts]: I am fine with removing it, so do `Option` and `Either` also not reexporting it.

**Comment by gcanti** [comment]
Thanks @mlegenhausen

**Pull request merged**